app/services/background.py

```
@@ -343,23 +394,58 @@ async def _autonomous_cycles_service(self):
+                    await asyncio.sleep(300)  # Wait 5 minutes before trying again
+                    continue
+                
+                # Check if any users are actually active
+                try:
+                    autonomous_keys = await self.redis.keys("autonomous_active:*")
+                    if len(autonomous_keys) == 0:
+                        self.logger.debug("No active autonomous users - skipping cycle")
+                        await asyncio.sleep(60)  # Short wait when no active users
+                        continue
+                        
+                except Exception as e:
```

Ruff (0.12.2)

Do not catch blind exception: `Exception`

(BLE001)